File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/studentWeb/src/contracts/learningGuideContract.ts
import type { EducationWorkspaceSection } from '../types/educationWorkspaceSection'

export type LearningGuideStepStatus =
  | 'COMPLETE'
  | 'NEXT'
  | 'AVAILABLE'
  | 'OPTIONAL'
  | 'BLOCKED'

export interface LearningGuideStepContract {
  contractName: 'LearningGuideStep.v1'
  section: Exclude<EducationWorkspaceSection, 'STUDENT'>
  title: string
  description: string
  status: LearningGuideStepStatus
  actionLabel: string
}

export interface LearningGuideContract {
  contractName: 'LearningGuide.v1'
  recommendedSection: Exclude<EducationWorkspaceSection, 'STUDENT'>
  headline: string
  message: string
  completedSteps: number
  totalSteps: number
  steps: LearningGuideStepContract[]
}

File: apps/backend/app/contracts/learningGuideContract.py
from typing import Literal
from pydantic import BaseModel


GuideSection = Literal[
    "ACADEMIC_STAGE",
    "LEARNING_CONTEXT",
    "SUBJECT",
    "LEARNING_UNIT",
    "MATERIAL",
    "PEDAGOGICAL",
    "LIA_TUTOR",
    "LEARNING_GOAL",
    "STUDY_SCOPE",
    "STUDY_SESSION",
    "LEARNING_PROGRESS",
]

GuideStepStatus = Literal[
    "COMPLETE",
    "NEXT",
    "AVAILABLE",
    "OPTIONAL",
    "BLOCKED",
]


class LearningGuideStepContract(BaseModel):
    contractName: Literal["LearningGuideStep.v1"] = "LearningGuideStep.v1"
    section: GuideSection
    title: str
    description: str
    status: GuideStepStatus
    actionLabel: str


class LearningGuideContract(BaseModel):
    contractName: Literal["LearningGuide.v1"] = "LearningGuide.v1"
    recommendedSection: GuideSection
    headline: str
    message: str
    completedSteps: int
    totalSteps: int
    steps: list[LearningGuideStepContract]
